modules/mcc/src/common.hpp

```diff
@@ -79,10 +79,10 @@ void polyanticlockwise(std::vector<T> &points)
     // Sort the points in anti-clockwise order
     // Trace a line between the first and second point.
     // If the third point is at the right side, then the points are anti-clockwise
-    cv::Point v1 = points[1] - points[0];
-    cv::Point v2 = points[2] - points[0];
+    T v1 = points[1] - points[0];
+    T v2 = points[2] - points[0];
 
-    double o = (v1.x * v2.y) - (v1.y * v2.x);
+    double o = ((double)v1.x * v2.y) - ((double)v1.y * v2.x);
```

Comment:
Why do you force the values to double. It could be done after computation to exclude compiler warnings. Also you can replace condition from `o < 0` to `(v1.x * v2.y) >= (v1.y * v2.x)` to not wary about expression return type.

Reply:
My reasoning was: `T` could be an integer type so `v1.x * v2.y` may overflow, so it is safer to cast to `double` earlier, since the result is stored in a `double o` anyway.

I looked at `polyanticlockwise()` and `polyclockwise()` uses and `T` is always `cv::Point2f`. I suggest we drop the `template` and `o`. What do you think?

Reply:
good idea